tests/e2e/demo-modal-compare.spec.ts

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable no-console -- we want to log when test fails */
+import { isTesting as isTestingEnvironment } from '@lmc-eu/spirit-common/constants/environments';
+import { SERVERS, getDevelopmentEndpointUri } from '@lmc-eu/spirit-common/constants/servers';
+import { expect, test } from '@playwright/test';
+
+const runComponentCompareTests = ({ componentsDir, packageName, componentName }) => {
+  if (!packageName) return;
+
+  const formattedPackageName = packageName
+    .split('-')
+    .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
+    .join(' ');
+
+  test.describe(`Demo ${formattedPackageName} Components`, () => {
+    test(`test demo ${formattedPackageName} component ${componentName}`, async ({ page }) => {
+      try {
+        const url = isTestingEnvironment()
+          ? SERVERS.TESTING[packageName]
+          : getDevelopmentEndpointUri(packageName, { isDocker: packageName !== 'web-twig' });
+
+        await page.goto(`${url}${componentsDir}/${componentName}/${packageName === 'web-twig' ? '?HIDE_TOOLBAR' : ''}`);
+        await waitForPageLoad(page);
+
+        await runModalTests(page, componentName);
+      } catch (error) {
+        console.error(`Test for demo ${formattedPackageName} component ${componentName} failed. ${error}`);
+        throw error;
+      }
+    });
+  });
+};
+
+const waitForPageLoad = async (page) => {
+  // Wait for fonts to load
+  await page.evaluate(() => document.fonts.ready);
+
+  // Wait for images to load
+  await page.waitForFunction(() => Array.from(document.querySelectorAll('img')).every((img) => img.complete));
+
+  // Wait for transitions to finish
+  await page.waitForLoadState();
+
+  // Disable animations to avoid flaky screenshots
+  await page.addStyleTag({ content: '*, *::before, *::after { animation-iteration-count: 1 !important }' });
+};
+
+const runModalTests = async (page, componentName) => {
+  // Helper function for taking and comparing screenshots
+  const takeScreenshot = async (suffix, fullPage = false) => {
+    await expect(page).toHaveScreenshot(`${componentName.toLowerCase()}-${suffix}.png`, {
+      animations: 'disabled',
+      fullPage,
+    });
+  };
+
+  // TEST: open basic modal, close with backdrop click
+  await page.click('[data-test-id="basic-modal"]');
```

Comment:
I wonder if we should somehow mention that we do these tests in the demos, so developers know they can't modify them freely. What do you think?

Reply:
Hmm, I'm not sure how to handle this... I don't want to add comments all over the code. Personally, I think the data attribute is self-explanatory and indicates that it's for testing purposes. If someone changes or deletes it, the tests will alert them. 😄